### PR TITLE
Feature lpdev 135 refactoring

### PR DESCRIPTION
**Pull request by aushakov**

### CODE REVIEW

File: client/Lykke.Service.PayInvoice.Client/Api/IInvoicesApi.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+{
+    internal interface IInvoicesApi
+    {
+        [Get("/api/merchants/{merchantId}/invoices")]
+        Task<IEnumerable<InvoiceModel>> GetAllAsync(string merchantId);
+        
+        [Get("/api/merchants/{merchantId}/invoices/{invoiceId}")]
+        Task<InvoiceModel> GetByIdAsync(string merchantId, string invoiceId);
+        
+        //[Post("/api/merchants/{merchantId}/invoices/drafts")]
```
Comment (tarurar) [inline client/Lykke.Service.PayInvoice.Client/Api/IInvoicesApi.cs:16]: better to delete comments?

File: client/Lykke.Service.PayInvoice.Client/PayInvoiceClient.cs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            var result = await _httpClient.PostAsync($"/api/merchants/{merchantId}/invoices", content);
+
+            if (result.IsSuccessStatusCode)
+            {
+                string value = await result.Content.ReadAsStringAsync();
+                return JsonConvert.DeserializeObject<InvoiceModel>(value);
+            }
+
+            throw new ErrorResponseException("An error occurred  during calling api");
+            //return await _runner.RunAsync(() => _invoicesApi.CreateAsync(merchantId, model));
```
Comment (tarurar) [inline client/Lykke.Service.PayInvoice.Client/PayInvoiceClient.cs:113]: Please remove comments

File: src/Lykke.Service.PayInvoice.Services/InvoiceService.cs [hunk 3]
```diff
@@ -1,272 +1,278 @@
...
+                    MarkupPips = 0,
+                    PaymentAssetId = invoice.PaymentAssetId,
+                    SettlementAssetId = invoice.SettlementAssetId
+                });
+        }
 
-                return response.Address;
-            }
-            catch (Exception exception)
+        // TODO: Rewrite
```
Comment (tarurar) [inline src/Lykke.Service.PayInvoice.Services/InvoiceService.cs:252]: Please create task in jira for this

File: src/Lykke.Service.PayInvoice/Controllers/EmployeesController.cs [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+                return NotFound(ErrorResponse.Create(exception.Message));
+            }
+            catch (Exception exception)
+            {
+                await _log.WriteErrorAsync(nameof(EmployeesController), nameof(GetAsync),
+                    merchantId.ToContext(nameof(merchantId))
+                        .ToContext(nameof(employeeId), employeeId)
+                        .ToJson(), exception);
+
+                throw;
```
Comment (tarurar) [inline src/Lykke.Service.PayInvoice/Controllers/EmployeesController.cs:90]: Wouldn't it be better to return 500 explicitly?

**Review by tarurar — APPROVED** [review]

File: src/Lykke.Service.PayInvoice/Controllers/EmployeesController.cs (hunk 4 above)
Comment (aushakov) [reply src/Lykke.Service.PayInvoice/Controllers/EmployeesController.cs:90]: Middleware will do it and log request details.

File: client/Lykke.Service.PayInvoice.Client/PayInvoiceClient.cs (hunk 2 above)
Comment (aushakov) [reply client/Lykke.Service.PayInvoice.Client/PayInvoiceClient.cs:113]: It's Okay, I hope Refit will be updated and I can use it this case

**Pull request merged**